context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +163,40 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
If `compareTo()` is going to throw when tickers are different, so should `isBefore()`